app/src/main/java/org/schabi/newpipe/local/playlist/LocalPlaylistFragment.java

```diff
@@ -331,6 +358,127 @@ public void onComplete() { }
+                            PlaylistStreamEntry playlistItem = playlistIter.next();
+                            int indexInHistory = Collections.binarySearch(historyStreamIds,
+                                    playlistItem.getStreamId());
+
+                            if (indexInHistory < 0) {
+                                notWatchedItems.add(playlistItem);
+                            } else if (!thumbnailVideoRemoved
+                                    && playlistManager.getPlaylistThumbnail(playlistId)
+                                    .equals(playlistItem.getStreamEntity().getThumbnailUrl())) {
+                                thumbnailVideoRemoved = true;
+                            }
+                        }
+                    } else {
+                        while (playlistIter.hasNext()) {
+                            PlaylistStreamEntry playlistItem = playlistIter.next();
+                            int indexInHistory = Collections.binarySearch(historyStreamIds,
+                                    playlistItem.getStreamId());
+
+                            boolean hasState = streamStatesIter.next() != null;
+                            if (indexInHistory < 0 ||  hasState) {
+                                notWatchedItems.add(playlistItem);
+                            } else if (!thumbnailVideoRemoved
+                                    && playlistManager.getPlaylistThumbnail(playlistId)
+                                    .equals(playlistItem.getStreamEntity().getThumbnailUrl())) {
+                                thumbnailVideoRemoved = true;
+                            }
+                        }
+                    }
+
+                    return Flowable.just(notWatchedItems, thumbnailVideoRemoved);
+                })
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(flow -> {
+                    List<PlaylistStreamEntry> notWatchedItems =
+                            (List<PlaylistStreamEntry>) flow.blockingFirst();
+                    boolean thumbnailVideoRemoved = (Boolean) flow.blockingLast();
+
+                    itemListAdapter.clearStreamItemList();
+                    itemListAdapter.addItems(notWatchedItems);
+                    saveChanges();
+
+
+                    if (thumbnailVideoRemoved) {
+                        updateThumbnailUrl();
+                    }
+
+                    long videoCount = itemListAdapter.getItemsList().size();
+                    setVideoCount(videoCount);
+                    if (videoCount == 0) {
+                        showEmptyState();
+                    }
+
+                    hideLoading();
+
+                    //If this is not done, 'removeWatchedDisposable', will never be disposed of.
+                    //Why: Because using the 'removePartiallyWatched' in this functions parms,
+                    // prevents it from disposing. Exact reason for this behavior is unknown
```

Comment:
They should all have a space after `//`. Also, there probably is a proper way of solving this.

Reply:
I have looked around, and I'm not sure what the solution would be.

Reply:
Fixed '// ' issue in 776dddd

Reply:
@GradyClark: You should just add this thing to the already existing `CompositeDisposable` in this class (`disposables`) and not do this stuff here at all, see e.g. [here](https://github.com/TeamNewPipe/NewPipe/blob/50046e2a3de419c0583ed6e8c2bc5dd26e13dd4f/app/src/main/java/org/schabi/newpipe/fragments/list/channel/ChannelFragment.java#L250).

Reply:
@wb9688 but then how can he check if the removal is already going on, and in that case do nothing? See [line 385](https://github.com/TeamNewPipe/NewPipe/pull/3065/files/db335d5cec3b77be37d897279973f1fbf94537af#diff-2ac03147edde19bf93d75ca5d662e7e2R385)

Reply:
@Stypox: Can't you create e.g. a `private boolean isRemovingWatched` for that?